tests/test_timeseriesx.py

```diff
@@ -817,6 +817,32 @@ def test_timestamp_series_fill_gaps():
     assert ts[timestamps[-1] + dt.timedelta(hours=1)] == 2
 
 
+def test_timestamp_series_fill_gaps_no_start_and_end():
+    timestamps = [
+        dt.datetime(2020, 3, 1, 15, 0, 0),
+        dt.datetime(2020, 3, 1, 16, 0, 0),
+        dt.datetime(2020, 3, 1, 17, 0, 0),
+    ]
+    values = [1, 1, np.nan]
+    ts = TimestampSeries.create_from_lists(timestamps, values, freq=pd.offsets.Hour())
+    ts = ts.fill_gaps(value=2)
+    assert len(ts) == 3
+    assert ts.values == [1, 1, 2]
+
+
+def test_timestamp_series_fill_gaps_with_unit(default_timestamp_series):
```

Comment:
this test actually covers two test cases. please split into two. 
* __first__: `test_timestamp_series_fill_gaps_with_start_end`:
    * do not focus on units here  
    * please assert also on the timestamps, not only the values
    * you may also provide a value for start
* __second__: `test_timestamp_series_fill_gaps_with_unit`:
    * test adding a dimensionless number to a unit based series
    * assert on integrity of the `_series`-dtype

Reply:
The first part is already implemented in the `test_timestamp_series_fill_gaps`, no? Or would you prefer additional checks? For the second one I added the dtype-check to the test

Reply:
There should be a test for each public method. Indeed, the `get_gaps` relies on `fill_gaps`. But i don't want to have tests based on assumptions on internals of the implementation of public methods, rather on their signature and expected behaviour.

Reply:
Sorry, I oversaw the existing tests.